File: Epoch-ScriptCore/Source/Epoch/Math/Noise.cs
using System;

namespace Epoch
{
    public static class Noise
    {
        public static void SetSeed(int aSeed) => InternalCalls.Noise_SetSeed(aSeed);
        public static float SimplexNoise(float x, float y) => InternalCalls.Noise_SimplexNoise(x, y);
        public static float PerlinNoise(float x, float y) => InternalCalls.Noise_PerlinNoise(x, y);
    }
}

File: Epoch/src/Epoch/Math/Noise.cpp
#include "epch.h"
#include "Noise.h"
#include "FastNoise/FastNoise.h"

namespace Epoch
{
	static FastNoise mySimplexFastNoise;
	static FastNoise myPerlinFastNoise;

	void Noise::SetSeed(int aSeed)
	{
		mySimplexFastNoise.SetSeed(aSeed);
		myPerlinFastNoise.SetSeed(aSeed);
	}

	float Noise::SimplexNoise(float x, float y)
	{
		return mySimplexFastNoise.GetNoise(x, y);
	}

	float Noise::PerlinNoise(float x, float y)
	{
		return myPerlinFastNoise.GetNoise(x, y);
	}
}